de.metas.business/src/main/java/org/adempiere/pricing/api/impl/CalculateDiscountCommand.java

```diff
@@ -73,21 +87,16 @@ else if (X_M_DiscountSchema.DISCOUNTTYPE_Formula.equals(discountType)
 		}
 
 		final I_M_DiscountSchemaBreak breakApplied = fetchDiscountSchemaBreak();
+
 		if (breakApplied != null)
 		{
-			final BigDecimal discount;
-			if (breakApplied.isBPartnerFlatDiscount())
-			{
-				discount = request.getBPartnerFlatDiscount();
-			}
-			else
+			if (breakApplied.isPriceOverride())
 			{
-				discount = breakApplied.getBreakDiscount();
+				return computeIndividualPriceForDiscountSchemaBreak(breakApplied);
 			}
-			return DiscountResult.builder()
-					.discount(discount)
-					.C_PaymentTerm_ID(breakApplied.getC_PaymentTerm_ID())
-					.build();
+
+			return computeDefaultDiscountForDiscoutSchemaBreak(breakApplied);
+
```

Comment:
You removed the support for BPartnerFlatDiscount but we still need it.
More, the DiscountResult shall be the merged result of:
* computing the prices if configured
* calculating the discount and payment term conditions like it was before